IdentityCore/src/JWEResponse/MSIDJweResponse+EcdhAesGcm.m

```diff
@@ -0,0 +1,251 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.  
+
+
+#import <Foundation/Foundation.h>
+#import "MSIDJweResponse+EcdhAesGcm.h"
+#import "MSIDJWECrypto.h"
+#import "MSIDJweResponse.h"
+#import "NSData+MSIDExtensions.h"
+#import "NSData+MSIDEccSecKeyRef.h"
+#import "MSIDEcdhApv.h"
+#import "MSIDJsonSerializer.h"
+#import "IdentityCore-Swift.h"
+#import "MSIDFlightManager.h"
+#import "MSIDConstants.h"
+
+MSIDJWECryptoKeyExchangeAlgorithm const MSID_KEY_EXCHANGE_ALGORITHM_ECDH_ES = @"ECDH-ES";
+MSIDJWECryptoKeyResponseEncryptionAlgorithm const MSID_RESPONSE_ENCRYPTION_ALGORITHM_A256GCM = @"A256GCM";
+
+@implementation MSIDJweResponse (EcdhAesGcm)
+
+- (nullable NSDictionary *)decryptJweResponseWithPrivateStk:(nonnull SecKeyRef)privateStkRef
+                                                  jweCrypto:(nonnull MSIDJWECrypto *)jweCrypto
+                                                      error:(NSError * _Nullable __autoreleasing * _Nullable)error
+{
+    // 1. Check for necessary request parameters
```

Comment:
should we print out request params (hide PII info) at INFO/DEBUG level to help debugging? Crypto operation always hard to debug if something wrong.